peripherals/spi/spi_v3.yaml

```diff
@@ -0,0 +1,281 @@
+"SPI*":
+  CR1:
+    IOLOCK:
+      Unlocked: [0, "IO configuration unlocked"]
+      Locked: [1, "IO configuration locked"]
+    TCRCI:
+      AllZeros: [0, "All zeros TX CRC initialization pattern"]
+      AllOnes: [1, "All ones TX CRC initialization pattern"]
+    RCRCI:
+      AllZeros: [0, "All zeros RX CRC initialization pattern"]
+      AllOnes: [1, "All ones RX CRC initialization pattern"]
+    CRC33_17:
+      Disabled: [0, "Full size (33/17 bit) CRC polynomial is not used"]
+      Enabled: [1, "Full size (33/17 bit) CRC polynomial is used"]
+    SSI:
+      SlaveSelected: [0, "0 is forced onto the SS signal and the I/O value of the SS pin is ignored"]
+      SlaveNotSelected: [1, "1 is forced onto the SS signal and the I/O value of the SS pin is ignored"]
+    HDDIR:
+      Receiver: [0, "Receiver in half duplex mode"]
+      Transmitter: [1, "Transmitter in half duplex mode"]
+    CSUSP:
+      _write:
+        NotRequested: [0, "Do not request master suspend"]
+        Requested: [1, "Request master suspend"]
+    CSTART:
+      NotStarted: [0, "Do not start master transfer"]
+      Started: [1, "Start master transfer"]
+    MASRX:
+      Disabled: [0, "Automatic suspend in master receive-only mode disabled"]
+      Enabled: [1, "Automatic suspend in master receive-only mode enabled"]
+    SPE:
+      Disabled: [0, "Peripheral disabled"]
+      Enabled: [1, "Peripheral enabled"]
+
+  CR2:
+    TSER: [0, 65535]
+    TSIZE: [0, 65535]
+
+  CFG1:
+    MBR:
+      Div2: [0, "f_spi_ker_ck / 2"]
+      Div4: [1, "f_spi_ker_ck / 4"]
+      Div8: [2, "f_spi_ker_ck / 8"]
+      Div16: [3, "f_spi_ker_ck / 16"]
+      Div32: [4, "f_spi_ker_ck / 32"]
+      Div64: [5, "f_spi_ker_ck / 64"]
+      Div128: [6, "f_spi_ker_ck / 128"]
+      Div256: [7, "f_spi_ker_ck / 256"]
+    CRCEN:
+      Disabled: [0, "CRC calculation disabled"]
+      Enabled: [1, "CRC calculation enabled"]
+    CRCSIZE: [3, 31]
```

Comment:
This is correct _but_ svd2rust will not generate any safe API to write to this field because it can't stop you writing 0-2 at runtime. One option would be to set this field to `[0, 31]`, which would permit invalid values 0-2 to be written, but I don't think that causes any more serious problem than any other misconfiguration of the peripheral (e.g. `CRCSIZE` must be a multiple of `DSIZE` which we cannot enforce). That way svd2rust generates a safe method to write integers to the field.

Reply:
ACK